src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs

```diff
@@ -25,12 +29,45 @@ internal sealed class RmdPublishHandler : IMarkdownFlavorPublishHandler {
             return true;
         }
 
-        public string GetCommandLine(string inputFile, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
-            string format = GetDocumentTypeString(publishFormat);
+        public async Task Publish(IRSession session, string inputFilePath, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
+            // Send the markdown to R-Host for processing as a blob
+            byte[] rmdData = File.ReadAllBytes(inputFilePath);
+            var rmdBlobId = await session.SendBlobAsync(rmdData);
+
+            // Perform markdown rendering on the blob
             string outputFile = Path.GetFileName(outputFilePath);
-            string outputFolder = Path.GetDirectoryName(outputFilePath).Replace('\\', '/');
-            // Run rmarkdown::render
-            return Invariant($"rmarkdown::render('{inputFile}', output_format='{format}', output_file='{outputFile}',  output_dir='{outputFolder}', encoding='cp{encoding.CodePage}')");
+            string format = GetDocumentTypeString(publishFormat);
+            string ext = Path.GetExtension(outputFilePath).Substring(1);
+            var publishResult = await session.EvaluateAsync($"rtvs:::rmarkdown_publish(blob_id = {rmdBlobId}, output_filename = '{outputFile}', output_format='{format}', encoding='cp{encoding.CodePage}')", REvaluationKind.Normal);
+            JObject obj = publishResult.Result as JObject;
```

Comment:
Use `(T)` for casting where the cast is not expected to fail (so that it throws early rather than producing `null`).